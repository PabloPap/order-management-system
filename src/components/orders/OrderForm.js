import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../shared/TextInput';
import SelectInput from '../shared/SelectInput';

const OrderForm = ({
  order,
  statusAll,
  onSave,
  onChange,
  saving = false,
  errors = {},
}) => {
  return (
    <form onSubmit={onSave}>
      <h2>{order.id ? 'Edit' : 'Add'} Order</h2>
      {errors.onSave && <p>{errors.onSave}</p>}
      <TextInput
        name="title"
        label="Title"
        value={order.title || ''}
        onChange={onChange}
        error={errors.title}
      />
      <TextInput
        name="orderNum"
        label="Order ID"
        value={order.orderNum || ''}
        onChange={onChange}
        error={errors.orderNum}
      />
      <SelectInput
        name="status"
        label="Status"
        value={order.status || ''}
        defaultOption="Select Status"
        options={statusAll.map((status) => ({
          value: status.id,
          text: status.name,
        }))}
        onChange={onChange}
        error={errors.status}
      />

      <button type="submit" disabled={saving}>
        {saving ? 'Saving...' : 'Save'}
      </button>
    </form>
  );
};

OrderForm.propTypes = {
  statusAll: PropTypes.array.isRequired,
  order: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool,
};

export default OrderForm;
