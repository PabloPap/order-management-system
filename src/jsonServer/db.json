{"orders":[{"id":1,"title":"Power tools","slug":"power-tools","orderId":249500,"status":1},{"id":2,"title":"Fishing equipment","slug":"fishing-equipment","orderId":149348,"status":1},{"id":3,"title":"Antique furniture","slug":"antique-furniture","orderId":495856,"status":3},{"id":4,"title":"Smart watch","slug":"smart-watch","orderId":617985,"status":2},{"id":5,"title":"Mountain bike","slug":"Mountain-bike","orderId":906854,"status":1},{"id":6,"title":"Hiking backpacks","slug":"hiking-backpacks","orderId":857343,"status":3},{"id":7,"title":"Folding wallets","slug":"folding-wallets","orderId":338495,"status":2},{"id":8,"title":"Gaming chair","slug":"gaming-chair","orderId":485908,"status":2},{"id":9,"title":"Smart tv","slug":"smart-tv","orderId":889586,"status":2},{"id":10,"title":"Power supply adapter","slug":"power-supply-adapter","orderId":908489,"status":1}],"statusAll":[{"id":1,"name":"Pending"},{"id":2,"name":"Completed"},{"id":3,"name":"Cancelled"}]}