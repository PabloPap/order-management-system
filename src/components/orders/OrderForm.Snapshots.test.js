import React from 'react';
import OrderForm from './OrderForm';
import renderer from 'react-test-renderer';
import { orders, statusAll } from '../../jsonServer/mockData';

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <OrderForm
      order={orders[0]}
      statusAll={statusAll}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />,
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <OrderForm
      order={orders[0]}
      statusAll={statusAll}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />,
  );

  expect(tree).toMatchSnapshot();
});
