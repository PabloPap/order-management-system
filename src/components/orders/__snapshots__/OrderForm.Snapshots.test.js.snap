// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[MockFunction]}
  >
    <h1>
      Edit
       Order
    </h1>
    <div
      className="form__group"
    >
      <input
        className=""
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Power tools"
      />
      <label
        className="form__group--label"
        htmlFor="title"
      >
        Title
      </label>
      <i
        className="form__group--bar"
      />
    </div>
    <div
      className="form__group"
    >
      <input
        className=""
        name="orderNum"
        onChange={[MockFunction]}
        type="text"
        value={249500}
      />
      <label
        className="form__group--label"
        htmlFor="orderNum"
      >
        Order ID
      </label>
      <i
        className="form__group--bar"
      />
    </div>
    <div
      className="form__group"
    >
      <select
        name="status"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Status
        </option>
        <option
          value={1}
        >
          Pending
        </option>
        <option
          value={2}
        >
          Completed
        </option>
        <option
          value={3}
        >
          Cancelled
        </option>
      </select>
      <label
        className="form__group--label"
        htmlFor="status"
      >
        Status
      </label>
      <i
        className="form__group--bar"
      />
    </div>
    <div
      className="form__button"
    >
      <button
        disabled={false}
        type="submit"
      >
        <span>
          Save
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[MockFunction]}
  >
    <h1>
      Edit
       Order
    </h1>
    <div
      className="form__group"
    >
      <input
        className=""
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Power tools"
      />
      <label
        className="form__group--label"
        htmlFor="title"
      >
        Title
      </label>
      <i
        className="form__group--bar"
      />
    </div>
    <div
      className="form__group"
    >
      <input
        className=""
        name="orderNum"
        onChange={[MockFunction]}
        type="text"
        value={249500}
      />
      <label
        className="form__group--label"
        htmlFor="orderNum"
      >
        Order ID
      </label>
      <i
        className="form__group--bar"
      />
    </div>
    <div
      className="form__group"
    >
      <select
        name="status"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Status
        </option>
        <option
          value={1}
        >
          Pending
        </option>
        <option
          value={2}
        >
          Completed
        </option>
        <option
          value={3}
        >
          Cancelled
        </option>
      </select>
      <label
        className="form__group--label"
        htmlFor="status"
      >
        Status
      </label>
      <i
        className="form__group--bar"
      />
    </div>
    <div
      className="form__button"
    >
      <button
        disabled={true}
        type="submit"
      >
        <span>
          Saving...
        </span>
      </button>
    </div>
  </form>
</div>
`;
